nix/shell-plugins.nix

```diff
@@ -0,0 +1,57 @@
+{ pkgs, lib, config, is-home-manager, ... }:
+with lib;
+let cfg = config.programs._1password-shell-plugins;
+in {
+  options = {
+    programs._1password-shell-plugins = {
+      enable = mkEnableOption "1Password Shell Plugins";
+      plugins = mkOption {
+        type = types.listOf types.package;
+        default = [ ];
+        example = literalExpression ''
+          with pkgs; [
+            gh
+            awscli2
+            cachix
+          ]
+        '';
+        description =
+          "CLI Packages to enable 1Password Shell Plugins for; ensure that a Shell Plugin exists by checking the docs: https://developer.1password.com/docs/cli/shell-plugins/";
+      };
+    };
+  };
+
+  config = let
```

Comment:
the `config` block is the Nix expression that gets generated based on what the user sets for the `options` we defined above.